storage/series/series.go

```diff
@@ -156,44 +155,56 @@ func (s *dbSeries) updateBlocksWithLock() updateBlocksResult {
 
 		result.ActiveBlocks++
 
+		if cachePolicy == CacheAll || retriever == nil {
+			// Never unwire
+			result.WiredBlocks++
+			continue
+		}
+
 		if !currBlock.IsRetrieved() {
+			// Already unwired
 			result.UnwiredBlocks++
 			continue
 		}
 
-		var unwired bool
-		switch {
-		case cachePolicy == CacheAll:
-			// Never unwire
-		case retriever != nil:
-			// Potentially unwire
-			sinceLastRead := now.Sub(currBlock.LastReadTime())
-			shouldUnwire := sinceLastRead >= wiredTimeout &&
-				retriever.IsBlockRetrievable(start)
-			if shouldUnwire {
-				switch cachePolicy {
-				case CacheAllMetadata:
-					// Keep the metadata but remove contents
-
-					// NB(r): Each block needs shared ref to the series ID
-					// or else each block needs to have a copy of the ID
-					id := s.id
-					metadata := block.RetrievableBlockMetadata{
-						ID:       id,
-						Length:   currBlock.Len(),
-						Checksum: currBlock.Checksum(),
-					}
-					currBlock.ResetRetrievable(start, retriever, metadata)
-				default:
-					// Remove the block and it will be looked up later
-					s.blocks.RemoveBlockAt(start)
-					currBlock.Close()
+		// Potentially unwire
+		var unwired, shouldUnwire bool
+		if retriever.IsBlockRetrievable(start) {
```

Comment:
Is the IsBlockRetrievable just making sure that its been flushed to disk?

Reply:
Correct yeah.